demisto_sdk/commands/upload/uploader.py

```diff
@@ -32,470 +67,238 @@ class Uploader:
+            parent_dir_name = get_parent_directory_name(self.path)
+            if parent_dir_name in UNIFIED_ENTITIES_DIR:
+                status_code = self.unified_entity_uploader(self.path) or status_code
             elif os.path.basename(self.path.rstrip('/')) in CONTENT_ENTITIES_DIRS:
-                self.directory_uploader(self.path.rstrip('/'))
-
-            # Input is a pack
+                status_code = self.entity_dir_uploader(self.path) or status_code
             elif parent_dir_name == PACKS_DIR:
-                self.pack_uploader()
+                status_code = self.pack_uploader(self.path) or status_code
+
+        if not self.successfully_uploaded_files and not self.failed_uploaded_files and \
+                not self.unuploaded_due_to_version:
+            # if not uploaded any file
+            click.secho(
+                f'\nError: Given input path: {self.path} is not uploadable. '
+                f'Input path should point to one of the following:\n'
+                f'  1. Pack\n'
+                f'  2. A content entity directory that is inside a pack. For example: an Integrations directory or '
+                f'a Layouts directory\n'
+                f'  3. Valid file that can be imported to Cortex XSOAR manually. '
+                f'For example a playbook: helloWorld.yml',
+                fg='bright_red'
+            )
+            return 1
+
+        print_summary(self.successfully_uploaded_files, self.unuploaded_due_to_version, self.failed_uploaded_files)
+        return status_code
+
+    def file_uploader(self, path: str) -> int:
+        """
+        Upload a file.
+        Args:
+            path: The path of the file to upload. The rest of the parameters are taken from self.
+
+        Returns:
 
-            # Input is not supported
+        """
+        try:
+            upload_object: Union[YAMLObject, JSONObject] = path_to_pack_object(path)
+        except ContentFactoryError:
+            file_name = os.path.split(path)[-1]
+            message = f"Cannot upload {path} as the file type is not supported for upload."
+            if self.log_verbose:
+                click.secho(message, fg='bright_red')
+            self.failed_uploaded_files.append((file_name, "Unknown", message))
+            return 1
+
+        file_name = upload_object.path.name  # type: ignore
+        entity_type = find_type(str(upload_object.path))
+
+        if entity_type in UPLOAD_SUPPORTED_ENTITIES:
+            if upload_object.from_version <= self.demisto_version <= upload_object.to_version:  # type: ignore
+                try:
+                    if entity_type in [FileType.INTEGRATION, FileType.SCRIPT, FileType.PLAYBOOK,
+                                       FileType.TEST_PLAYBOOK]:
+                        result = upload_object.upload(self.client)  # type: ignore
+                    else:
+                        result = upload_object.upload(self.client)  # type: ignore
```

Comment:
If you always use `upload_object.upload()` why is the if statement necessary?

Reply:
It was there for the override parameter. Will merge into one